File: game.py
class Cell:
  def __init__(self, letter):
    self.letter = letter
    self.visited = False

  def __str__(self):
    return self.letter

class Game:
  def __init__(self):
    self.board = []
    self.wordLengths = []

  def __str__(self):
    ret = ""
    for row in self.board:
      for cell in row:
        ret = ret + str(cell) + " "
      ret = ret + "\n"
    return ret

  def buildBoardFromFile(self, filename):
    with open(filename) as f:
      row = -1
      for line in f:
        line = line.rstrip()
        if row == -1:
          self.wordLengths = sorted(line.split(","), reverse=True)
          self.wordLengths = [int(x) for x in self.wordLengths]
        else:
          self.board.append([])
          column = 0
          for item in line:
            self.board[row].append(Cell(item))
            column = column + 1
        row = row + 1

  def getHeight(self):
    return len(self.board)

  def getWidth(self):
    return len(self.board[0])

File: js/game.js
var Cell = function(letter, row, col) {
  this.letter = letter.toUpperCase();
  this.visited = false;
  this.row = row;
  this.col = col;
}

Cell.prototype.setVisited = function(row, col, visited) {
  this.visited = visited;
  var message = new WorkerMessage(
      WorkerMessage.Code.SET_VISITED,
      {"row": row, "col": col, "visited": visited});
  Util.postMessage(message);
}

var Game = function(lengths) {
  this.board = [];
  this.wordLengths = lengths;
  for (length in this.wordLengths) {
     this.wordLengths[length] = parseInt(this.wordLengths[length]);
   }
}

Game.prototype.buildBoard = function(input) {
  var lines = input.split('\n');
  for (row in lines) {
    this.board.push([]);
    for (col in lines[row]) {
      this.board[row].push(new Cell(lines[row][col], row, col));
    }
  }
}

Game.prototype.getHeight = function() {
  return this.board.length;
}

Game.prototype.getWidth = function() {
  return this.board[0].length;
}

Game.prototype.setInWords = function(nodes) {
  var coords = [];
  for (i in nodes) {
    coords.push([nodes[i].row, nodes[i].col]);
  }
  var message = new WorkerMessage(
      WorkerMessage.Code.SET_IN_WORDS,
      {'coords': coords});
  Util.postMessage(message);
}

Game.prototype.clearInWords = function(nodes) {
  var coords = [];
  for (i in nodes) {
    coords.push([nodes[i].row, nodes[i].col]);
  }
  var message = new WorkerMessage(
      WorkerMessage.Code.CLEAR_IN_WORDS,
      {'coords': coords});
  Util.postMessage(message);
}

Game.prototype.printAnswer = function(words) {
  var message = new WorkerMessage(
      WorkerMessage.Code.PRINT_ANSWER,
      {'words': words});
  Util.postMessage(message);
}